### ISSUE TITLE
2D Plans problem

### ISSUE THREAD

**Issue by matilde-mantoan-efm** [ThatOpen/engine_components#450]
### Describe the bug 📝

Hi! We are developing a IFC viewer with Angular 14. We have implemented the Plans Component, but we have noticed that in case of many models the component doesn't behave as expected:

3D with 3 loaded models:
![image](https://github.com/user-attachments/assets/e7848d82-1bd4-461a-8e70-272f48ba4849)

This is plan "L03" with 3 loaded models:
![image](https://github.com/user-attachments/assets/4b79163a-aa6d-4059-9f5d-1ac47c5b6554)

3D with 8 loaded models:
![image](https://github.com/user-attachments/assets/3786c9d4-ae30-40b6-9290-201ff1008dac)

2D - plan "L03" with 8 loaded models:
![image](https://github.com/user-attachments/assets/8f70f093-8440-47ac-bc24-4cd8f5dc2f75)

Up to 2/3 loaded models it works perfectly, but with 8 models the Plans component produces multiple overlapped plans.

### System Info 💻

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 5.45 GB / 15.73 GB
  Binaries:
    Node: 18.13.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.19.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (126.0.2592.102)
    Internet Explorer: 11.0.19041.4355
```

### Used Package Manager 📦

npm

### Validations ✅

- [X] Read the [docs](https://docs.thatopen.com/intro).
- [X] Check that there isn't [already an issue](https://github.com/ThatOpen/engine_components/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a repository issue and not a framework-specific issue. For example, if it's a THREE.js related bug, it should likely be reported to [mrdoob/threejs](https://github.com/mrdoob/three.js) instead.
- [X] Check that this is a concrete bug. For Q&A join our [Community](https://people.thatopen.com/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by agviegas** [comment]
Hey @matilde-mantoan-efm can you please send those models to antonio@thatopen.com to check it out? Thanks!

**Comment by RyugaRyuzaki** [comment]
@matilde-mantoan-efm @agviegas i think this came from [this line](https://github.com/ThatOpen/engine_components/blob/v1.5.0/src/fragments/FragmentHighlighter/index.ts#L396) 
- Because when we load multi model, FragmentsManager [changed the second model reference to first model](https://github.com/ThatOpen/engine_components/blob/main/packages/core/src/fragments/FragmentsManager/index.ts#L215-L238)
- So i change looks like [this]( to 
```
    const tempMesh = new THREE.Mesh(mesh.geometry);
    tempMesh.matrix.copy(mesh.matrix);
    const tempMatrix = new THREE.Matrix4();
    instanced.getMatrixAt(i, tempMatrix);
    tempMesh.applyMatrix4(tempMatrix);
    tempMesh.applyMatrix4(mesh.matrix);

    if (mesh.parent) tempMesh.applyMatrix4(mesh.parent.matrix);
    tempMesh.updateMatrix();
    tempMesh.updateMatrixWorld();
```
and it works for me

**Comment by RyugaRyuzaki** [comment]
Btw @agviegas what do you think if we  change [this interface](https://github.com/ThatOpen/engine_components/blob/main/packages/front/src/core/ClipEdges/src/types.ts#L42-L51) to
```
export interface Edge {
  /** The name of the style to which this Edges belong. */
  name: string;

  /** The visible clipping lines in the scene. */
  meshes: {[uuid:string]:THREE.LineSegments};
  /** The fill of the section. */
  fill?: ClippingFills;
}
```
Because of when i use streamer 
```
 /**
     * detect whenever model is loaded
     * storage walls
     */
    fragmentStreamerLoader.onModelLoaded.add((model: FRAGS.FragmentsGroup) => {
      classifier.byEntity(model);
      const wallClassifier = classifier.find({
        entities: FragmentPlans.wallEntities,
      });
      const doorClassifier = classifier.find({
        entities: FragmentPlans.doorEntities,
      });
      for (const fragID in wallClassifier) {
        this.walls.add(fragID);
        filled.fragments[fragID] = new Set(wallClassifier[fragID]);
      }
      for (const fragID in doorClassifier) {
        this.doors.add(fragID);
        projected.fragments[fragID] = new Set(doorClassifier[fragID]);
      }
    });
    /**
     * detect whenever fragment is loaded
     */
    fragmentStreamerLoader.onFragmentsLoaded.add((frags) => {
      const filledSize = filled.meshes.size;
      const projectedSize = projected.meshes.size;
      for (const frag of frags) {
        if (this.walls.has(frag.id)) {
          filled.meshes.add(frag.mesh);
        }
        if (this.doors.has(frag.id)) {
          projected.meshes.add(frag.mesh);
        }
      }
      const update =
        filledSize < filled.meshes.size ||
        projectedSize < projected.meshes.size;
      if (update) this.updatePlans();
    });
```
As i understand everytime fragmentStreamerLoader.onFragmentsLoaded we also update plan so [this function](https://github.com/ThatOpen/engine_components/blob/main/packages/front/src/core/ClipEdges/src/clipping-edges.ts#L177)  will re-calculate

**Comment by agviegas** [comment]
@RyugaRyuzaki your idea is to have one edge per fragment, so that when a new fragment is loaded, only that edge is recalculated?

**Comment by RyugaRyuzaki** [comment]
Yes. That what I mean. However some issues with performance if we create many line segment in the the scene

**Comment by agviegas** [comment]
@RyugaRyuzaki yes I was thinking that. It will increase the draw calls dramatically. Not sure it's a wise move regarding performance

**Comment by agviegas** [comment]
@RyugaRyuzaki was right about [this](https://github.com/ThatOpen/engine_components/issues/450#issuecomment-2249839914). This is available from `@thatopen/components-front@2.1.15`. 

Before:

![image](https://github.com/user-attachments/assets/d0f65751-7b49-416c-aa78-be120c7e1a8f)

After:

![image](https://github.com/user-attachments/assets/a3fa481c-f46d-4514-bb93-e5991dedfcea)

**Comment by matilde-mantoan-efm** [comment]
Thank you!!
